java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrow.java

```diff
@@ -32,15 +31,57 @@
 public class AvroToArrow {
 
   /**
-   * Fetch the data from {@link GenericDatumReader} and convert it to Arrow objects.
+   * Fetch the data from {@link Decoder} and convert it to Arrow objects.
    * @param schema avro schema.
+   * @param decoder avro decoder
    * @param allocator Memory allocator to use.
    * @return Arrow Data Objects {@link VectorSchemaRoot}
    */
   public static VectorSchemaRoot avroToArrow(Schema schema, Decoder decoder, BaseAllocator allocator)
```

Comment:
I think we should make this package private, and comment that it is only meant for testing.